pkg/assets/store.go

```diff
@@ -220,6 +220,10 @@ func (s *Store) addToken(ctx context.Context, ns string, sel v1.SecretKeySelecto
 }
 
 func (s *Store) AddBearerToken(ctx context.Context, ns string, sel v1.SecretKeySelector, key string) error {
+	if sel == (v1.SecretKeySelector{}) {
+		return nil
+	}
+
```

Comment:
Sorry for being unclear, I meant to change the signature of AddBearerToken() like this:
```suggestion
func (s *Store) AddBearerToken(ctx context.Context, ns string, sel *v1.SecretKeySelector, key string) error {
	if sel == nil {
		return nil
	}

```

Reply:
Hi if I set it that way I get an error `invalid operation: sel == nil (mismatched types "k8s.io/api/core/v1".SecretKeySelector and untyped nil)compiler[MismatchedTypes](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#MismatchedTypes)`

Reply:
hmm this commit should work https://github.com/simonpasquier/prometheus-operator/commit/a94c876861773f182f5fd3845ede028d9dc40ffc

Reply:
Well in that change you are chaning the func addToken instead of AddBearerToken. So could you please clarify?

Reply:
yes I suggest to change both function signatures.

Reply:
Hi @simonpasquier done, when you have the time pls review